### PR TITLE
Feature/#59 Added review questions in the room

### PR DESCRIPTION
**Pull request by developerdevpav**

### CODE REVIEW

File: Backend/Interview.Domain/Events/DatabaseProcessors/RoomEvaluationPostProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public override async ValueTask ProcessModifiedAsync(RoomQuestionEvaluation original, RoomQuestionEvaluation current, CancellationToken cancellationToken)
+    {
+        await HandleEvent(current, cancellationToken, (roomId, questionId) =>
+        {
+            var evaluationPayload = new RoomEvaluationChangeEventPayload(questionId);
+            return new RoomEvaluationChangeEvent(roomId, evaluationPayload);
+        });
+    }
+
+    private async ValueTask HandleEvent<T>(RoomQuestionEvaluation entity, CancellationToken cancellationToken, Func<Guid, Guid, T> funcEventMapper)
```
Comment (blowin) [inline Backend/Interview.Domain/Events/DatabaseProcessors/RoomEvaluationPostProcessor.cs:39]: Let's discuss today.

Comment (developerdevpav) [reply Backend/Interview.Domain/Events/DatabaseProcessors/RoomEvaluationPostProcessor.cs:39]: Ok.

File: Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/IRoomQuestionEvaluationRepository.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace Interview.Domain.Rooms.RoomQuestionEvaluations;
+
+public interface IRoomQuestionEvaluationRepository : IRepository<RoomQuestionEvaluation>
+{
+    Task<RoomQuestionEvaluation?> FindByActiveQuestionRoomAsync(Guid roomId, Guid userId, CancellationToken cancellationToken);
+
+    Task<RoomQuestionEvaluation?> FindByQuestionIdAndRoomAsync(Guid roomId, Guid questionId, Guid userId, CancellationToken cancellationToken);
+
+    Task Submit(Guid roomId, Guid userId, CancellationToken cancellationToken);
```
Comment (blowin) [inline Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/IRoomQuestionEvaluationRepository.cs:11]: SubmitAsync

File: Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/Permissions/RoomQuestionEvaluationPermissionAccessor.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private readonly ISecurityService _securityService;
+    private readonly IRoomQuestionEvaluationService _roomQuestionEvaluationService;
+
+    public RoomQuestionEvaluationPermissionAccessor(IRoomQuestionEvaluationService roomQuestionEvaluationService, ISecurityService securityService)
+    {
+        _securityService = securityService;
+        _roomQuestionEvaluationService = roomQuestionEvaluationService;
+    }
+
+    public async Task<QuestionEvaluationDetail> FindByRoomIdAndQuestionId(QuestionEvaluationGetRequest request, CancellationToken cancellationToken)
```
Comment (blowin) [inline Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/Permissions/RoomQuestionEvaluationPermissionAccessor.cs:19]: FindByRoomIdAndQuestionIdAsync

File: Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/Records/Response/QuestionEvaluationDetail.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+namespace Interview.Domain.Rooms.RoomQuestionEvaluations.Records.Response;
+
+public class QuestionEvaluationDetail
+{
+    public QuestionEvaluationDetail(Guid id)
+    {
+        Id = id;
+    }
+
+    public Guid Id { get; }
```
Comment (blowin) [inline Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/Records/Response/QuestionEvaluationDetail.cs:10]: public **required** Guid Id { get; init; }

File: Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/SERoomQuestionEvaluationState.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+public class SERoomQuestionEvaluationState : SmartEnum<SERoomQuestionEvaluationState>
+{
+    public static readonly SERoomQuestionEvaluationState Draft = new("Draft", 0, EVRoomQuestionEvaluationState.Draft);
+    public static readonly SERoomQuestionEvaluationState Submitted = new("Submitted", 1, EVRoomQuestionEvaluationState.Submitted);
+
+    public EVRoomQuestionEvaluationState EnumValue { get; }
+
+    private SERoomQuestionEvaluationState(string name, int value, EVRoomQuestionEvaluationState enumValue)
+        : base(name, value)
```
Comment (blowin) [inline Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/SERoomQuestionEvaluationState.cs:13]: : base(name, (int)enumValue)

File: Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/SERoomQuestionEvaluationState.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+namespace Interview.Domain.Rooms.RoomQuestionEvaluations;
+
+public class SERoomQuestionEvaluationState : SmartEnum<SERoomQuestionEvaluationState>
+{
+    public static readonly SERoomQuestionEvaluationState Draft = new("Draft", 0, EVRoomQuestionEvaluationState.Draft);
+    public static readonly SERoomQuestionEvaluationState Submitted = new("Submitted", 1, EVRoomQuestionEvaluationState.Submitted);
+
+    public EVRoomQuestionEvaluationState EnumValue { get; }
+
+    private SERoomQuestionEvaluationState(string name, int value, EVRoomQuestionEvaluationState enumValue)
```
Comment (blowin) [inline Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/SERoomQuestionEvaluationState.cs:12]: value can be removed

File: Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/SERoomQuestionEvaluationState.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        : base(name, value)
+    {
+        EnumValue = enumValue;
+    }
+
+    public static SERoomQuestionEvaluationState? FromEnum(EVRoomQuestionEvaluationState enumValue)
+    {
+        return List.FirstOrDefault(it => it.EnumValue == enumValue);
```
Comment (blowin) [inline Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/SERoomQuestionEvaluationState.cs:20]: FromValue((int)enumValue)

File: Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/Services/RoomQuestionEvaluationService.cs [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    private async Task<RoomQuestionEvaluation?> FindQuestionEvaluation(Guid roomId, Guid questionId, Guid userId, CancellationToken cancellationToken)
+    {
+        await ValidateQuestionEvaluation(roomId, userId, cancellationToken);
+
+        return await _roomQuestionEvaluationRepository.FindByQuestionIdAndRoomAsync(roomId, questionId, userId, cancellationToken);
+    }
+
+    private async Task ValidateQuestionEvaluation(Guid roomId, Guid userId, CancellationToken cancellationToken)
```
Comment (blowin) [inline Backend/Interview.Domain/Rooms/RoomQuestionEvaluations/Services/RoomQuestionEvaluationService.cs:88]: We should think about separating this logic into a separate component, there are a lot of checks like this.

**Pull request merged**